client/src/extension.ts

```diff
@@ -116,16 +118,82 @@ export function activate(context: ExtensionContext) {
         // nothing more to do
         return;
     }
+
+    // First check recursively all the files set in `project.kao`-like files, starting with `project.kao`.
+    workspace.findFiles(`**/project.kao`).then((uris) => {
+        uris.forEach((uri) => {
+            openProjectFile(uri);
+            // There should be only one `project.kao` file.
+            return;
+        });
+    });
+
     /*
-     * List of all the yseopml file extensions known by this extension as set in `client/package.json`.
+     * List of all the yseopml file extensions known by this extension as set in `client/package.json`
+     * but never explictly set in project.kao-like files.
      */
-    const yseopmlExtensions = ['kao', 'yclass', 'yobject', 'ycomplete', 'dcl', 'yml'];
-
+    const yseopmlExtensions = ['yclass', 'yobject', 'ycomplete'];
     for (const extension of yseopmlExtensions) {
         parseFilesWithExtension(extension);
     }
 }
 
+/**
+ * Try to open a file with URI `fileUri`.
+ * Then, if the file is a `project.kao`-like file (i.e. is a list files used for the project),
+ * read its content and apply this function recursivly for each line that are existing file's URI.
+ *
+ * @param fileUri An existing file URI.
+ */
+function openProjectFile(fileUri: Uri): void {
+    // Try to open the file. If it is opened, the server will parse it.
+    workspace.openTextDocument(fileUri).then(
+        // The document exists and was successfully opened and should be parsed already.
+        (doc) => {
+            console.debug(`Parsing ${fileUri}.`);
+            const lines = doc.getText().split('\n');
+            if (lines.length === 0 || !lines[0].trim().startsWith('_FILE_TYPE_')) {
+                // We are not in a `project.kao`-like file. Stop the going further.
+                return;
+            }
+            lines.forEach((line) => {
+                // line can be indented in the file.
+                line = line.trim();
+
+                /*
+                 * Ignore:
+                 * - any empty line
+                 * - lines that are just preprocessing stuff or Yseop Engine instruction.
+                 * - every file from .generated-yml/ directory
+                 */
+                if (
+                    line.length === 0 ||
+                    line.indexOf('@') >= 0 ||
+                    line.indexOf('_FILE_TYPE_') >= 0 ||
+                    line.indexOf('.generated-yml') >= 0
```

Comment:
That’s a bit prone to false positives. If you’ve got time, consider something like

### Skip

```properties
.generated-yml
.generated-yml/
./.generated-yml
foo/.generated-yml
foo/.generated-yml/bar
```

### Parse

```properties
foo/plop.generated-yml
foo/.generated-yml.patate/bar
[…]
```

You’ll probably hate me for suggesting to make something that stupid more robust, though. But to me, this more or less deserves a function of its own rather than an `indexOf`. xD

Reply:
Well, in our case none of them will be parsed.
And I clearly don't want to handle such false positive right know because of their almost zero probability in YML projects.

Reply:
> in our case none of them will be parsed

That’s the point, hehe.

> almost zero probability

That’s the kind of answer I expected, but the current implem still strikes me as an incredibly sloppy way to handle paths. With a proper object type, you could loop through the path segments and find whether one of them is `.generated-yml` in a flash.

Reply:
Done more or less now. I didn't find an object type to handle path segments in JS/TS.